File: tests/fixtures/textmate/php/basic.php
declare(strict_types=1);

/**
 * Small grammar fixture with café, 東京, λ, 🚀, and astral 𝌆.
 */
#[Attribute]
final class Greeting {
    public function __construct(
        public readonly string $name,
        private array $tags = ['basic', 'unicode'],
    ) {}

    public function render(int $count = 1): string {
        $label = match ($count) {
            0 => 'none',
            1 => "café {$this->name} 🚀 𝌆",
            default => "{$count} visits to 東京",
        };
        return $label . ': ' . implode(',', $this->tags);
    }
}

$greeting = new Greeting(name: 'Mark');
echo $greeting->render(count: 2);

File: tests/fixtures/textmate/ruby/basic.rb
# frozen_string_literal: true
=begin
Ruby basic fixture: café, 東京, λ, 🚀, and astral 𝌆.
=end
module Aurora
  PATTERN = /\A(?:ready|launch)\z/i
  BANNER = <<~TEXT
    Mission café 東京 🚀 𝌆
  TEXT

  class Greeting
    attr_reader :name, :tags
    def initialize(name:, tags: %i[basic unicode])
      @name = name
      @tags = tags
    end

    def render(count = 1)
      label = case count
              when 0 then "none"
              when 1 then "hello #{@name}"
              else "#{count} launches for #{@name}"
              end
      "#{label}: #{@tags.join(', ')}"
    end
  end
end

greeter = Aurora::Greeting.new(name: "λ")
puts [1, 2].map { |count| greeter.render(count) }
